common/mqtt_client/tests/rumqttd_broker.rs

```diff
@@ -0,0 +1,64 @@
+use std::{
+    collections::HashMap,
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4},
+};
+
+use librumqttd::{async_locallink, Config, ConnectionSettings, ConsoleSettings, ServerSettings};
+use port_scanner::scan_port;
+
+pub async fn start_broker_local(port: u16) -> anyhow::Result<()> {
+    if !scan_port(58585) {
```

Comment:
I believe this should use port function parameter.

Reply:
```rust
if tokio::net::TcpStream::connect(("0.0.0.0", port)).await.is_ok() {
...
}
```

Reply:
Adding a dependency, even dev, for a one-line function is indeed a bit luxury.

Reply:
yes, fixed